src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -94,12 +96,35 @@ private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorIn
                 $this->logger->debug('Calling getCredentials() on guard configurator.', array('firewall_key' => $this->providerKey, 'authenticator' => get_class($guardAuthenticator)));
             }
 
+            // abort the execution of the authenticator if it doesn't supports the request.
+            if ($guardAuthenticator instanceof GuardAuthenticatorInterface) {
+                if (true !== $guardAuthenticator->supports($request)) {
+                    return;
+                }
+                // as there was a support for given request,
+                // authenticator is expected to give not-null credentials.
+                $credentialsCanBeNull = false;
+            } else {
+                // it's a GuardAuthenticatorLegacyInterface
```

Comment:
We should trigger the deprecation notice here. For example: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/SecurityContext.php#L14